recap/types.py

```diff
@@ -539,7 +539,10 @@ def clean_dict(type_dict: dict | list | str) -> dict | list | str:
     :param type_dict: A type dictionary, list, or string.
     :return: A cleaner, more compact type dictionary, list, or string.
     """
-
+    if isinstance(type_dict, dict) and "type" not in type_dict:
+        raise ValueError(
+            "'type' is a required field and was not found in the dictionary."
+        )
```

Comment:
wdyt of `elif` on the next if block below?

Reply:
Fine with me! Check out my latest change and let me know if I misunderstood.